tests/unit/adapters/application-test.js

```diff
@@ -224,51 +226,27 @@ test('#findRelated', function(assert) {
 test('#findRelated is called with optional type for the resource', function (assert) {
   assert.expect(4);
   const done = assert.async();
-  let supervisor = this.container.lookup('model:supervisor').create({
-    type: 'supervisors',
-    id: '1000000',
-    attributes: {
-      name: 'The Boss',
-    },
-    relationships: {
-      employees: {
-        links: {
-          related: 'http://api.pixelhandler.com/api/v1/supervisors/1000000/employees'
-        }
-      }
-    }
-  });
 
-  let EmployeeAdapter = Adapter.extend({type: 'employees', url: '/employees'});
-  this.registry.register('service:supervisors', EmployeeAdapter.extend({
-    cacheLookup: function() {
-      return supervisor;
-    }
+  let supervisor = this.container.lookup('model:supervisor').create(supervisorMock.data);
+  let employee = this.container.lookup('model:employee').create(employeeMock.data);
+
+  let SupervisorAdapter = Adapter.extend({type: 'supervisors', url: '/supervisors'});
+  SupervisorAdapter.reopenClass({isServiceFactory: true});
+  let EmployeeAdapter   = Adapter.extend({type: 'employees', url: '/employees'});
+  EmployeeAdapter.reopenClass({isServiceFactory: true});
+
+  this.registry.register('service:employees', EmployeeAdapter.extend({
+    cacheLookup: function () { return employee; }
   }));
-  EmployeeAdapter.reopenClass({ isServiceFactory: true });
-  this.registry.register('service:employees', EmployeeAdapter.extend());
-  let service = this.container.lookup('service:employees');
-  let stub = sandbox.stub(service, 'findRelated', function () {
-    return RSVP.Promise.resolve(supervisor);
-  });
-  let employee = this.container.lookup('model:employee').create({
-    type: 'employees',
-    id: '1000001',
-    attributes: {
-      name: 'The Special',
-    },
-    relationships: {
-      supervisor: {
-        links: {
-          related: 'http://api.pixelhandler.com/api/v1/employees/1000001/supervisor'
-        }
-      }
-    }
+  let employeeService = this.container.lookup('service:employees');
+  let stub = sandbox.stub(employeeService, 'findRelated', function () {
+    return RSVP.Promise.resolve(null);
   });
-  let url = employee.get('relationships.supervisor.links.related');
-  employee.get('supervisor').then(() => {
+
+  let url = supervisor.get('relationships.direct-reports.links.related');
+  supervisor.get('directReports').then(() => {
     assert.ok(stub.calledOnce, 'employees service findRelated method called once');
-    assert.equal(stub.firstCall.args[0].resource, 'supervisor', 'findRelated called with supervisor resource');
+    assert.equal(stub.firstCall.args[0].resource, 'direct-reports', 'findRelated called with "direct-reports" resource');
```

Comment:
@aars I like using 'direct-reports' better than using 'supervisor' this improves the test.